spec/lib/rewriters/logical_operator_assignment_spec.rb

```diff
@@ -0,0 +1,202 @@
+require 'lib/spec_helper'
+
+describe Opal::Rewriters::LogicalOperatorAssignment do
+  def s(type, *children)
+    ::Opal::AST::Node.new(type, children)
+  end
+
+  let(:rewriter) { Opal::Rewriters::LogicalOperatorAssignment.new }
+
+  def parse(source)
+    parser = Opal::Parser.default_parser
+    buffer = ::Parser::Source::Buffer.new('(eval)')
+    buffer.source = source
+    parser.parse(buffer)
+  end
+
+  def rewrite(ast)
+    rewriter.process(ast)
+  end
+
+  around(:each) do |e|
+    Opal::Rewriters::LogicalOperatorAssignment::AbstractHandler.reset_tmp_counter!
+    Opal::Rewriter.disable { e.run }
+  end
+  let(:cache_tmp_name) { :$or_equals_lhs_recvr_tmp_1 }
+  let(:cached) { s(:js_tmp, cache_tmp_name) }
+
+  shared_examples 'it rewrites' do |from:, to:|
+    it "rewrites #{from.inspect} to #{to.inspect}" do
+      input = parse(from)
+      rewritten = rewrite(input)
+      expected = parse(to)
+
+      expect(rewritten).to eq(expected)
+    end
+  end
+
+  context 'rewriting or_asgn and and_asgn nodes' do
+    context 'local variable' do
+      include_examples 'it rewrites', from: 'a = local; a ||= 1', to: 'a = local; a || a = 1'
```

Comment:
No, it should be `a || a= 1`. Assignment happens only if `a` is nil. I've found it today when I was implementing a case for a `send` node. Writer gets called only when reader returns `nil`.